phone/src/apps/messages/components/modal/MessageImageModal.tsx

```diff
@@ -57,17 +64,21 @@ export const MessageImageModal = ({ isOpen, messageGroupId, onClose, image }: IP
   }, [image, sendFromQueryParam]);
 
   const sendFromClipboard = (event) => {
-    if (event.key === 'Enter') {
-      sendImageMessage(message);
-    }
+    navigator.clipboard.readText().then((text) => {
+      if (isImage(text)) {
+        sendImageMessage(text);
```

Comment:
You could `setQueryParamImagePreview(text)` so it triggers the modal first to preview before you send, and the rest is already taken care after, because the preview modal has the sendImage action onClick.

Reply:
(BTW, `queryParamImagePreview` and `setQueryParamImagePreview` seems like a bad name now we should put something more generic like `imagePreview`, `setImagePreview`)